src/particles/pusher/ExternalFields.H

```diff
@@ -30,15 +30,21 @@ void ApplyExternalField(
     amrex::ParticleReal& ExmByp,
     amrex::ParticleReal& EypBxp,
     amrex::ParticleReal& Ezp,
-    const amrex::Real ext_ExmBy_slope,
-    const amrex::Real ext_Ez_slope,
-    const amrex::Real ext_Ez_uniform)
+    amrex::ParticleReal& Bxp,
+    amrex::ParticleReal& Byp,
+    amrex::ParticleReal& Bzp,
+    const amrex::RealVect& ext_E_uniform,
+    const amrex::RealVect& ext_B_uniform,
+    const amrex::RealVect& ext_E_slope,
+    const amrex::RealVect& ext_B_slope)
 {
-    if ((ext_ExmBy_slope == 0.) && (ext_Ez_slope == 0.) && (ext_Ez_uniform == 0.)) return;
 
-    ExmByp += ext_ExmBy_slope * xp;
-    EypBxp += ext_ExmBy_slope * yp;
-    Ezp += ext_Ez_uniform + ext_Ez_slope * zp;
+    ExmByp += ext_E_uniform[0] + ext_E_slope[0] * xp;
+    EypBxp += ext_E_uniform[1] + ext_E_slope[1] * yp;
```

Comment:
Don‘t you need to use By and Bx here as well?

Reply:
The Doxygen and documentation clarify it, this is the modified E field as in the simulation. I could also get the E field separately, but as long as the documentation is clear I think it's fine. Besides, it makes it consistent between code, input and output.

Reply:
The Doxygen and documentation clarify it, this is the modified E field as in the simulation. I could also get the E field separately, but as long as the documentation is clear I think it's fine. Besides, it makes it consistent between code, input and output.

Reply:
Any preference? Should we let the user specify the actual Ex/Ey fields instead?

Reply:
The clearer name `hipace.external_dx_ExmBy_dy_EypBx_dz_Ez` is quite long so maybe the current name is fine.